### PR TITLE
Add repository controller part 1

### PR DESCRIPTION
**Pull request by dpk31999**
## Related Tickets:
- [#Ticket ID](https://edu-redmine.sun-asterisk.vn/issues/xxxx)

## WHAT this PR does?
Add repository controller part 1
## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Comment by dpk31999** [comment]
ready

File: app/Http/Controllers/Admin/ImageProductController.php [hunk 1]
```diff
@@ -40,18 +47,16 @@ public function create()
...
-        ]);
+        try {
+            $this->productImageRepository->storeImageOfProduct($request->all(), $product->id);
 
-        return redirect()->route('admin.products.edit', $product->id)
-                        ->with('message', trans('products.message_add_image_success'));
+            return redirect()->route('admin.products.edit', $product->id)
+                ->with('message', trans('products.message_add_image_success'));
+        } catch (Throwable $th) {
+            dd($th);
```
Comment (quoctd-0607) [inline app/Http/Controllers/Admin/ImageProductController.php]: remove

**Comment by dpk31999** [comment]
ready

File: app/Repositories/Category/CategoryRepository.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        $category = $this->findOrFail($id);
+
+        $category->suggestProducts()->delete();
+
+        foreach ($category->products as $product) {
+            $product->ratings()->detach();
+            $product->images()->delete();
+            $product->orderDetails()->detach();
+            $product->favoriteProducts()->detach();
+        }
```
Comment (ghost) [inline app/Repositories/Category/CategoryRepository.php]: N+1 nè bn ơi :) load products nhé

**Comment by dpk31999** [comment]
ready

**Pull request merged**